kf/bpf.go

```diff
@@ -548,63 +549,107 @@ func (b *BPF) GetArtifacts(conf *config.Config) error {
 		return fmt.Errorf("download failed: %w", err)
 	}
 	defer resp.Body.Close()
-
 	buf := &bytes.Buffer{}
 	buf.ReadFrom(resp.Body)
-
 	if resp.StatusCode != http.StatusOK {
 		return fmt.Errorf("get request returned unexpected status code: %d (%s), %d was expected\n\tResponse Body: %s", resp.StatusCode, http.StatusText(resp.StatusCode), http.StatusOK, buf.Bytes())
 	}
 
-	archive, err := gzip.NewReader(buf)
-	if err != nil {
-		return fmt.Errorf("failed to create Gzip reader: %w", err)
-	}
-	defer archive.Close()
+	if strings.HasSuffix(kfRepoURL.String(), ".zip") {
+		bk := bytes.NewReader(buf.Bytes())
+		archive, err := zip.NewReader(bk, int64(bk.Len()))
+		if err != nil {
+			panic(err)
```

Comment:
```suggestion
			return fmt.Errorf("failed to create Zip reader: %w", err)
```
for consistency with line 604